pkg/artifactory/resource_artifactory_api_key.go

```diff
@@ -61,14 +61,13 @@ func resourceApiKeyRead(d *schema.ResourceData, m interface{}) error {
 
 	apiKey, resp, err := c.V1.Security.GetApiKey(context.Background())
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
 		return err
 	}
```

Comment:
The way the middleware is setup, anything >= 400 is considered an error. This actually simplifies code everywhere else because, frankly, the error code is generally irrelevant. Error is error

With the above in mind, I can't do this. It's only an error _if_ resty produced an error `AND` the responseCode was 404.

However, you did bring my attention to something: I am using the property and not the function for status code. I think I will make that change